From aeda04439e615da71c61d9ffab020741b7149d8a Mon Sep 17 00:00:00 2001
From: Jason Gerecke <killertofu@gmail.com>
Date: Thu, 10 Aug 2017 15:34:29 -0700
Subject: [PATCH 3/4] Add support for 2nd-generation Intuos Pro

This tablet is usable without this patch, but the device type is missing
the WCM_RING flag. This prevents the ring valuator from being initialized
correctly, and in turn causes recent versions of gnome-control-center to
not allow configuration of the ring.

https://bugzilla.gnome.org/show_bug.cgi?id=785358

Signed-off-by: Jason Gerecke <jason.gerecke@wacom.com>
Reviewed-by: Ping Cheng <ping.cheng@wacom.com>
(cherry picked from commit 1a2a23c2df0ec0fd0bf46432db58f32914a6a09f)
---
 src/wcmValidateDevice.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/wcmValidateDevice.c b/src/wcmValidateDevice.c
index bfff8e7..8eac0dd 100644
--- a/src/wcmValidateDevice.c
+++ b/src/wcmValidateDevice.c
@@ -202,6 +202,8 @@ int wcmDeviceTypeKeys(InputInfoPtr pInfo)
 			TabletSetFeature(priv->common, WCM_DUALRING | WCM_LCD);
 			/* fall through */
 
+		case 0x357: /* Intuos Pro 2 M */
+		case 0x358: /* Intuos Pro 2 L */
 		case 0x314: /* Intuos Pro S */
 		case 0x315: /* Intuos Pro M */
 		case 0x317: /* Intuos Pro L */
-- 
2.13.6

